plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -265,6 +265,35 @@ public void testCharVarcharComparison()
                 .hasStackTraceContaining("Unsupported type: char(3)");
     }
 
+    @Test
+    public void testAddColumnToPartitionedTable()
```

Comment:
Could you please add tests that will check:
- if spark can read table with added columns
- if optimize/vacuum works after adding columns ?

Reply:
> if spark can read table with added columns

Should I wait #11565?

Reply:
Verifying, even locally if that works would be great imho.
Regarding test - depends how long you would have to wait. 
@findinpath what is ETA for https://github.com/trinodb/trino/pull/11565 ?

Reply:
#11565 is in a good shape now. I am assuming that the PR will be ready to be merged in the following days.

Reply:
@ebyhr so my advice would be to wait for it and write a proper test

Reply:
Does it mean you guys want to have this PR on hold until the other one lands?

Reply:
I could read new columns from Spark, but it prints below error at the firs access. There's no error from the 2nd access though. I'm little wondering if it needs fix or not. 
```sql
trino> ALTER TABLE delta.default.test ADD COLUMN x int;
ADD COLUMN

spark-sql> SELECT * FROM default.test;
ERROR DeltaLog: Change in the table id detected while updating snapshot.

1	NULL

spark-sql> SELECT * FROM default.test;
1	NULL
```

Reply:
I don't like this error but it looks like it returns the correct result anyway. It looks like spark cached something. What image did you use for testing? Can you try modyfing it like this https://github.com/trinodb/trino/commit/087868977cc7729621123bbc445384c9dfcae30f ?

Reply:
I tested on this product test PR #11565 

@findinpath may want to disable the cache in https://github.com/trinodb/trino/pull/11565/files#diff-8ecb8972788b7b6221c837540b00b020c2404ef64548ae563e02a9b1a66c8d14

Reply:
@ebyhr could you check that disabling the cache will actually help with this ?

Reply:
would running `REFRESH TABLE` in Spark help?

Reply:
@homar @findepi Disabling the cache didn't help. `REFRESH TABLE` logs the same error "ERROR DeltaLog: Change in the table id detected..." and subsequent `SELECT` don't show the error.

JFYI: The error comes from
https://github.com/delta-io/delta/blob/728bf902542077ce1c2e97ca67a53c53bb460c64/core/src/main/scala/org/apache/spark/sql/delta/SnapshotManagement.scala#L574-L575

Reply:
It seems Spark reuse the same `metaData.id` when adding a new column.

```
trino> ALTER TABLE delta.default.test ADD COLUMN c4 int;
```

s3://presto-ci-test/test/_delta_log/00000000000000000004.json
```json
{"commitInfo":{"version":4,"timestamp":1653357796162,"userId":"yuya.ebihara","userName":"yuya.ebihara","operation":"ALTER TABLE","operationParameters":{"queryId":"20220524_020316_00004_imnpb"},"clusterId":"trino-dev-ffffffff-ffff-ffff-ffff-ffffffffffff","readVersion":0,"isolationLevel":"WriteSerializable","blindAppend":true}}
{"protocol":{"minReaderVersion":1,"minWriterVersion":2}}
{"metaData":{"id":"24b68017-cb79-4f0b-8ee7-e72a496bbaf4","format":{"provider":"parquet","options":{}},"schemaString":"{\"fields\":[{\"metadata\":{},\"name\":\"c1\",\"nullable\":true,\"type\":\"integer\"},{\"metadata\":{},\"name\":\"c2\",\"nullable\":true,\"type\":\"integer\"},{\"metadata\":{},\"name\":\"c3\",\"nullable\":true,\"type\":\"integer\"},{\"metadata\":{},\"name\":\"c4\",\"nullable\":true,\"type\":\"integer\"}],\"type\":\"struct\"}","partitionColumns":[],"configuration":{},"createdTime":1653357796162}}
```

```
spark-sql> ALTER TABLE default.test ADD COLUMN (x5 int);
```

s3://presto-ci-test/test/_delta_log/00000000000000000005.json
```json
{"commitInfo":{"timestamp":1653357938234,"operation":"ADD COLUMNS","operationParameters":{"columns":"[{\"column\":{\"name\":\"x5\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}}]"},"readVersion":4,"isBlindAppend":true,"operationMetrics":{}}}
{"metaData":{"id":"24b68017-cb79-4f0b-8ee7-e72a496bbaf4","format":{"provider":"parquet","options":{}},"schemaString":"{\"type\":\"struct\",\"fields\":[{\"name\":\"c1\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c2\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c3\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c4\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"x5\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}","partitionColumns":[],"configuration":{},"createdTime":1653357796162}}
```